ui-tests/tst/software/aws/toolkits/jetbrains/uitests/fixtures/ActionMenu.kt

```diff
@@ -0,0 +1,33 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.uitests.fixtures
+
+import com.intellij.remoterobot.RemoteRobot
+import com.intellij.remoterobot.data.RemoteComponent
+import com.intellij.remoterobot.fixtures.ComponentFixture
+import com.intellij.remoterobot.fixtures.FixtureName
+import com.intellij.remoterobot.search.locators.byXpath
+import com.intellij.remoterobot.utils.waitFor
+
+fun RemoteRobot.actionMenu(text: String): ActionMenuFixture {
+    val xpath = byXpath("text '$text'", "//div[@class='ActionMenu' and @text='$text']")
+    waitFor {
+        findAll<ActionMenuFixture>(xpath).isNotEmpty()
+    }
+    return findAll<ActionMenuFixture>(xpath).first()
+}
+
+fun RemoteRobot.actionMenuItem(text: String): ActionMenuItemFixture {
+    val xpath = byXpath("text '$text'", "//div[@class='ActionMenuItem' and @text='$text']")
```

Comment:
i wonder if there's a nice way to figure out the correct xpath and then just write some wrappers around that

Reply:
In what way? What would you like to see wrapped?

Reply:
More of a comment on "how annoying is it to craft these xpaths?"

The tool you demo'd looks good enough, but only concern is the stability of these expressions across versions and IDE variants